### PR TITLE
Apply two-step owership changing flow

### PR DESCRIPTION
**Pull request by charlesjhongc**
Applies to
- contracts/PermanentStorage.sol
- contracts/LimitOrder.sol
- contracts/UserProxy.sol

### CODE REVIEW

File: contracts/LimitOrder.sol [hunk 1]
```diff
@@ -78,11 +79,19 @@ contract LimitOrder is ILimitOrder, BaseLibEIP712, SignatureValidator, Reentranc
...
-        emit TransferOwnership(_newOperator);
+        emit OperatorNominated(_newOperator);
+    }
+
+    function acceptOwnership() external {
+        require(msg.sender == nominatedOperator, "LimitOrder: not nominated");
+        emit OperatorChanged(operator, nominatedOperator);
+
+        operator = nominatedOperator;
+        nominatedOperator = address(0);
```
Comment (pilagod) [inline contracts/LimitOrder.sol:94]: Not sure if we need time lock here as well? Although splitting into two steps, if no time lock, user is still in the risky situation.

**Review by changwu-tw — APPROVED** [review]
Same as we did before

File: contracts/LimitOrder.sol (hunk 1 above)
Comment (NIC619) [reply contracts/LimitOrder.sol:94]: I think this is mainly to prevent operator error.

**Review by NIC619 — APPROVED** [review]

File: contracts/LimitOrder.sol (hunk 1 above)
Comment (charlesjhongc) [reply contracts/LimitOrder.sol:94]: Yeah, this prevent incorrect operator to be assigned.

**Pull request merged**